File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var arraySign = function(nums) {
    let negative = 0
    for (let i = 0; i < nums.length; i++){
        if (nums[i] < 0){
            negative += 1
        }
        if (nums[i] == 0){
            return 0
        }
    }
    if (negative % 2 == 0){
        return 1
    }else if(negative % 2 ==1){
        return -1
    } 
};

File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.java
class Solution {
    public static int signFunc(int product){
        if (product > 0){
            return 1;
        } else if (product < 0) {
            return -1;
        }
        return 0;
    }
    public static int arraySign(int[] nums) {
        int product = 1;
        for (int num : nums){
            if (num > 0){
                product *= 1;
            } else if (num < 0) {
                product *= -1;
            }else {
                product *= 0;
            }
        }
        return signFunc(product);
    }
}